clientv3/client.go

```diff
@@ -99,6 +100,16 @@ func (c *Client) Ctx() context.Context { return c.ctx }
 // Endpoints lists the registered endpoints for the client.
 func (c *Client) Endpoints() []string { return c.cfg.Endpoints }
 
+// AvailableEndpoints returns all available endpoints in the cluster.
+// This must be called with AutoSync option.
+func (c *Client) AvailableEndpoints() []string {
```

Comment:
Probably drop this; I don't think it's much different from making a memberlist rpc call and having functions that only work when certain options like this are enabled is sort of confusing.